### PR TITLE
Convert sprite_move to method

### PR DESCRIPTION
**Pull request by Gymnasiast**

### CODE REVIEW

File: src/openrct2/world/Sprite.cpp [hunk 1]
```diff
@@ -682,25 +682,26 @@ static void SpriteSpatialMove(SpriteBase* sprite, const CoordsXY& newLoc)
  * @param z (dx)
  * @param sprite (esi)
  */
-void sprite_move(int16_t x, int16_t y, int16_t z, SpriteBase* sprite)
+void SpriteBase::MoveTo(const CoordsXYZ& newLocation)
 {
-    if (!map_is_location_valid({ x, y }))
+    auto loc = newLocation;
+    if (!map_is_location_valid(newLocation))
```
Comment (Gymnasiast) [inline src/openrct2/world/Sprite.cpp:688]:
```suggestion
    if (!map_is_location_valid(loc))
```

File: src/openrct2/peep/Guest.cpp [hunk 2]
```diff
@@ -3687,7 +3687,7 @@ void Guest::UpdateRideAdvanceThroughEntrance()
             actionZ += RideData5[ride->type].z;
         }
 
-        MoveTo((*loc).x, (*loc).y, actionZ);
+        MoveTo({ loc->x, loc->y, actionZ });
```
Comment (tupaschoal) [inline src/openrct2/peep/Guest.cpp:3690]: Can't see on cellphone if loc is XY or XYZ. If first then it can be { *loc, actionZ }

File: src/openrct2/peep/Guest.cpp [hunk 3]
```diff
@@ -4039,7 +4039,7 @@ void Guest::UpdateRideApproachVehicle()
 {
     if (auto loc = UpdateAction())
     {
-        MoveTo((*loc).x, (*loc).y, z);
+        MoveTo({ loc->x, loc->y, z });
```
Comment (tupaschoal) [inline src/openrct2/peep/Guest.cpp:4042]: Same as before

File: src/openrct2/peep/Guest.cpp [hunk 4]
```diff
@@ -4361,7 +4361,7 @@ void Guest::UpdateRideApproachExit()
...
```
Comment (tupaschoal) [inline src/openrct2/peep/Guest.cpp:4364]: Same

File: src/openrct2/peep/Guest.cpp [hunk 5]
```diff
@@ -4387,12 +4387,12 @@ void Guest::UpdateRideInExit()
             int16_t actionZ = ride->stations[current_ride_station].GetBaseZ();
 
             actionZ += RideData5[ride->type].z;
-            MoveTo((*loc).x, (*loc).y, actionZ);
+            MoveTo({ loc->x, loc->y, actionZ });
```
Comment (tupaschoal) [inline src/openrct2/peep/Guest.cpp:4390]: Same (and it goes on, k won't flood issues)

File: src/openrct2/peep/Guest.cpp [hunk 6]
```diff
@@ -4706,7 +4706,7 @@ void Guest::UpdateRideOnSpiralSlide()
                 newLocation.x += _SpiralSlideEnd[dir].x;
                 newLocation.y += _SpiralSlideEnd[dir].y;
 
-                MoveTo(newLocation.x, newLocation.y, z);
+                MoveTo({ newLocation, z });
```
Comment (Gymnasiast) [inline src/openrct2/peep/Guest.cpp:4709]: newLocation is a CoordsXY, so this should be ok.

File: src/openrct2/world/SpriteBase.h [hunk 7]
```diff
@@ -1,6 +1,7 @@
 #pragma once
 
 #include "../common.h"
+#include "Location.hpp"
```
Comment (tupaschoal) [inline src/openrct2/world/SpriteBase.h:4]: Move this to the cpp and just forward declare `CoordsXYZ` here.
```suggestion

class CoordsXYZ;
```

Comment (Gymnasiast) [reply src/openrct2/world/SpriteBase.h:4]: What is the advantage of that?

Comment (tupaschoal) [reply src/openrct2/world/SpriteBase.h:4]: You move the include to the source file and if you ever change `Location.hpp` content, you won't trigger recompilation of everyone that includes `SpriteBase`.

File: src/openrct2/actions/StaffHireNewAction.hpp [hunk 8]
```diff
@@ -227,7 +227,7 @@ DEFINE_GAME_ACTION(StaffHireNewAction, GAME_COMMAND_HIRE_NEW_STAFF_MEMBER, Staff
                 // NOTE: This state is required for the window to act.
                 newPeep->state = PEEP_STATE_PICKED;
 
-                sprite_move(newPeep->x, newPeep->y, newPeep->z, newPeep);
+                newPeep->MoveTo({ newPeep->x, newPeep->y, newPeep->z });
                 invalidate_sprite_2(newPeep);
```
Comment (duncanspumpkin) [inline src/openrct2/actions/StaffHireNewAction.hpp:231]: A peep MoveTo contains the invalidate function as well. So this can be removed.

File: src/openrct2/actions/StaffHireNewAction.hpp [hunk 9]
```diff
@@ -332,7 +332,7 @@ DEFINE_GAME_ACTION(StaffHireNewAction, GAME_COMMAND_HIRE_NEW_STAFF_MEMBER, Staff
             }
         }
 
-        sprite_move(x, y, z + 16, newPeep);
+        newPeep->MoveTo({ x, y, z + 16 });
         invalidate_sprite_2(newPeep);
```
Comment (duncanspumpkin) [inline src/openrct2/actions/StaffHireNewAction.hpp:336]: so can this

File: src/openrct2/rct1/S4Importer.cpp [hunk 10]
```diff
@@ -1246,7 +1246,7 @@ class S4Importer final : public IParkImporter
         dst->mini_golf_current_animation = src->mini_golf_current_animation;
         dst->mini_golf_flags = src->mini_golf_flags;
 
-        sprite_move(src->x, src->y, src->z, dst);
+        dst->MoveTo({ src->x, src->y, src->z });
         invalidate_sprite_2(dst);
```
Comment (duncanspumpkin) [inline src/openrct2/rct1/S4Importer.cpp:1250]: just a thought but we don't need to do any invalidation during an import as the whole screen will be redrawn after load.

File: src/openrct2/ride/Ride.cpp [hunk 11]
```diff
@@ -1083,11 +1083,11 @@ void ride_remove_peeps(Ride* ride)
                 if (peep->GetNextIsSloped())
                     newLoc.z += COORDS_Z_STEP;
                 newLoc.z++;
-                sprite_move(newLoc.x, newLoc.y, newLoc.z, peep);
+                peep->MoveTo(newLoc);
```
Comment (duncanspumpkin) [inline src/openrct2/ride/Ride.cpp:1086]: The invalidate statements on L1078 and L1094 can be removed.

File: src/openrct2/ride/Vehicle.cpp [hunk 12]
```diff
@@ -4522,13 +4522,13 @@ void Vehicle::UpdateMotionBoatHire()
 
                     if (!(curRide->boat_hire_return_direction & 1))
                     {
-                        uint16_t bp = curY & 0x1F;
-                        if (bp == 16)
+                        uint16_t tilePart = curY % COORDS_XY_STEP;
```
Comment (duncanspumpkin) [inline src/openrct2/ride/Vehicle.cpp:4525]: try keep unrelated changes to a diffrent PR
